test/sys/test_select.rs

```diff
@@ -0,0 +1,55 @@
+use nix::sys::select::*;
+use nix::unistd::{pipe, write};
+use nix::sys::signal::SigSet;
+use nix::sys::time::{TimeSpec, TimeValLike};
+use std::os::unix::io::RawFd;
+
+#[test]
+pub fn test_pselect() {
+    let (r1, w1) = pipe().unwrap();
+    write(w1, b"hi!").unwrap();
+    let (r2, _w2) = pipe().unwrap();
+
+    let mut fd_set = FdSet::new();
+    fd_set.insert(r1);
+    fd_set.insert(r2);
+
+    let timeout = TimeSpec::seconds(10);
+    let sigmask = SigSet::empty();
+    assert_eq!(
+        1,
+        pselect(None, &mut fd_set, None, None, &timeout, &sigmask).unwrap()
+    );
+    assert!(fd_set.contains(r1));
+    assert!(!fd_set.contains(r2));
+}
+
+#[test]
+pub fn test_pselect_nfds2() {
+    let _mtx = ::SIGNAL_MTX
```

Comment:
You put `_mtx` in the wrong test; only `test_pselect` modifies the sigmask.

Reply:
🤦🏻‍♂️  oh no